src/gui/folderman.cpp

```diff
@@ -1230,11 +1230,12 @@ void FolderMan::setDirtyNetworkLimits()
     }
 }
 
-void FolderMan::trayOverallStatus(const QList<Folder *> &folders,
-    SyncResult::Status *status, bool *unresolvedConflicts)
+void FolderMan::trayOverallStatus(const QList<Folder *> &folders, SyncResult::Status &status, bool &unresolvedConflicts, QDateTime &lastSyncDone)
 {
-    *status = SyncResult::Undefined;
-    *unresolvedConflicts = false;
+    status = SyncResult::Undefined;
```

Comment:
For reasons I usually also prefer to use pointers for out parameter, it kind of indicates this parameter is special...
I don't like functions that modify my passed reference....

Reply:
Maybe just return a tuple or a struct rather than out parameters?

Reply:
References are however safer to use. Passing a mutable reference also suggests it may be modified. I think something like this should be documented, though.

Using a tuple would work in this case, as we don't require the caller to create any of the parameters. Creating a little "return value class" would also do the job, but I'll go for the tuple.

Reply:
great, documenting that parameters might get modified is only good in theory.
1) It requires someone to write the doc
2) It requires someone to read the doc....